pkg/types/vsphere/validation/platform.go

```diff
@@ -94,3 +99,42 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port.
+func ValidateVCenterAddress(vCenter string) error {
+	if strings.TrimSpace(vCenter) == "" {
+		return errors.New("must specify the name of the vCenter")
+	}
+	vCenterURL, err := url.ParseRequestURI(vCenter)
```

Comment:
I have used ParseRequestURI() to check for url scheme. 
It does not return an error if there is a scheme present in the url string.
eg: https:forest.com
https://forest.com 
Both the above examples would not throw an error. 
"https:forest.com" returns an empty host and "https://forest.com" returns "forest.com" as host. 

In the later part of the method I am using url.Parse() to check for params and path because url.ParseRequestURI() would throw an error if a scheme is not mentioned and would possibly make it difficult to test other things.